helm/cluster-api-provider-azure/templates/deployment.yaml

```diff
@@ -0,0 +1,93 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    aadpodidbinding: capz-controller-aadpodidentity-selector
+    control-plane: capz-controller-manager
+    {{- include "labels.common" . | nindent 4 }}
+  name: capz-controller-manager
+  namespace: capz-system
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      control-plane: capz-controller-manager
+      {{- include "labels.selector" . | nindent 6 }}
+  template:
+    metadata:
+      annotations:
+        kubectl.kubernetes.io/default-logs-container: manager
+      labels:
+        aadpodidbinding: capz-controller-aadpodidentity-selector
+        control-plane: capz-controller-manager
+        {{- include "labels.common" . | nindent 8 }}
+    spec:
+      containers:
+      - args:
+        - --metrics-addr=127.0.0.1:8080
+        - --enable-leader-election
+        - --feature-gates=MachinePool={{ .Values.featuregates.machinepool }},AKS={{ .Values.featuregates.aks }}
+        env:
+        - name: AZURE_ENVIRONMENT
+          valueFrom:
+            configMapKeyRef:
+              key: cloud-env
+              name: capz-manager-bootstrap-cloud-env
+        - name: AZURE_SUBSCRIPTION_ID
+          valueFrom:
+            secretKeyRef:
+              key: "azure.azureoperator.subscriptionid"
+              name: credential-default
+        - name: AZURE_TENANT_ID
+          valueFrom:
+            secretKeyRef:
+              key: "azure.azureoperator.tenantid"
+              name: credential-default
+        - name: AZURE_CLIENT_ID
+          valueFrom:
+            secretKeyRef:
+              key: "azure.azureoperator.clientid"
+              name: credential-default
+        - name: AZURE_CLIENT_SECRET
+          valueFrom:
+            secretKeyRef:
+              key: "azure.azureoperator.clientsecret"
+              name: credential-default
+        - name: NODE_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: spec.nodeName
+        - name: POD_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.name
+        - name: POD_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        image: "{{ .Values.image.registry }}/{{ .Values.image.name }}:{{ .Values.image.tag }}"
+        imagePullPolicy: Always
```

Comment:
is always really necessary? Maybe we can relax it to `IfNotPresent` ?

Reply:
I'll add a note to the Release and reliability board. We should automate how we generate Helm charts, so we don't have to manually check for changes when there is a new version in upstream.